File: packages/sdk-csharp/src/Theatrical.Sdk/Types/Site.cs
namespace Theatrical.Sdk.Types;

public sealed record Site
{
    public required string Id { get; init; }
    public required string Name { get; init; }
    public string? Address { get; init; }
    public string? City { get; init; }
    public string? Country { get; init; }
    public double? Latitude { get; init; }
    public double? Longitude { get; init; }
    public string? Timezone { get; init; }
    public bool IsActive { get; init; } = true;
    public int? ScreenCount { get; init; }
    public string[]? Features { get; init; }
}

public sealed record Screen
{
    public required string Id { get; init; }
    public required string Name { get; init; }
    public int? Capacity { get; init; }
    public string[]? Formats { get; init; }
    public bool IsAccessible { get; init; }
}

File: packages/sdk-py/theatrical/types/site.py
"""Site (cinema location) types."""

from __future__ import annotations

from typing import Optional

from theatrical.types.base import ApiModel


class GeoLocation(ApiModel):
    latitude: float
    longitude: float


class Address(ApiModel):
    line1: str
    line2: Optional[str] = None
    city: str
    state: Optional[str] = None
    postal_code: str
    country: str


class Screen(ApiModel):
    id: str
    name: str
    seat_count: int
    formats: list[str]
    is_accessible: bool


class Amenity(ApiModel):
    id: str
    label: str
    icon: Optional[str] = None


class SiteConfig(ApiModel):
    booking_lead_time: int
    max_tickets_per_order: int
    loyalty_enabled: bool
    fnb_enabled: bool


class Site(ApiModel):
    id: str
    name: str
    address: Address
    location: GeoLocation
    screens: list[Screen]
    config: SiteConfig
    timezone: str
    currency: str
    is_active: bool
    amenities: Optional[list[Amenity]] = None
